Poppler rewrite

MickeyVI opened the pull request.

Commit:
#33 add sentence-join script, remaining moses and models

Commit:
#33 add sentence-join script, remaining moses and models

Commit:
#33 add sentence-join installation

Commit:
#35, #36 : prevent call start() multiple time in one language id +
prevent invalid _workerStatus + read stderr for prevent deadlock.

Commit:
#35: prevent call start() multiple time in one language id
#37: fix to remove sorting for getting max
#39,#40: remove fontweight because unused it
#41: change link pattern

lpla reviewed (changes requested):
Please, modify sentence-join.py, its documentation and its call from Java (SentenceJoin.java) so the paths to the kenlm binaries could be provided by parameter. For example, `sentence-join.py --kenlm_path "/home/user/kenlm/bin"`.

I would be also good to do that to the other paths in `PDFExtract.json`, this way we can automate all paths when installing Bitextor, instead of requesting the user to modify all these paths inside JSON and Python files.

Commit:
#34 error handling
#36 using thread to read stdout and redirect stderr to stdout
#43 add new input parameters --kenlm_path for execute sentence-join.py
#38 using thread to read stdout and stderr
#42 using stdout to get
#43 add config path to the kenlm binaries
#44 remove not used code

Commit:
#45, #46 fixed

lpla reviewed (changes requested):
Great! Before I integrate this code and release a new Python wrapper version, we need that 'sentece_join' and 'kenlm_path' settings could be also set through `java -jar PDFExtract.jar `, as those paths could be a pain to set by editing the `PDFExtract.json` when installing the Python wrapper (as each user is going to have different 'home' paths).

Commit:
#42 use stdin for pdftohtml

Commit:
#43 add arguments for sentence join and kenlm path

Proyag committed:
Fix reading threshold as float from file

`threshold` was previously being read as a string instead of a float, which was breaking its usage from PDFExtract.java

lpla committed:
Merge pull request #49 from Proyag/poppler-rewrite

Fix reading threshold as float from file

ramoelee committed:
#51 To prevent spurious warnings about sentenceJoin models.

ramoelee committed:
#50 Set RedirectErrorStream to false to avoid error: This binary file
contains Trie with Quantization and array-compressed pointers.
